nengo_gui/components/value.py

```diff
@@ -59,6 +59,9 @@ def update_client(self, client):
         # while there is data that should be sent to the client
         while len(self.data) > 0:
             item = self.data.popleft()
+            while len(self.data) > 0:
```

Comment:
I can't follow you there. To me all three variants look like they (almost, see below) exactly the same thing. If this is not the case, it should be made clear why this isn't the case, so that no one comes along and modifies the code in the way I proposed.

Also, I just noticed that the double while loop might introduce a race condition. I assume data is added to `self.data` concurrently? In that case it might be possible, that data is added after the inner while loop finished, but before the condition of the outer while loop is checked. In that case the code would enter the outer while loop again, overwriting `item` and thus it loses some data.